pkg/ruler/storage/instance/instance.go

```diff
@@ -207,6 +207,9 @@ func newInstance(cfg Config, reg prometheus.Registerer, logger log.Logger, newWa
 }
 
 func (i *Instance) Storage() storage.Storage {
+	i.mut.Lock()
```

Comment:
do we need synchronization for reading `i.storage`? I see it's initialized during `Run` but not mutated anywhere else.

Reply:
The race detector caught this one and the `Ready()` one because, without a mutex, goroutines can read this value while it's being written